app/services/facility_stats_service.rb

```diff
@@ -28,25 +31,32 @@ def add_facility_stats(facility_data)
     size = facility_data.region.source.facility_size
     add_size_section(size) unless stats_by_size[size]
     periods.each do |period|
-      current_period = stats_by_size[size][period]
+      current_period = stats_by_size[size][:periods][period]
       current_period[rate_numerator] += facility_data.dig(rate_numerator, period) || 0
-      current_period["adjusted_registrations"] += facility_data["adjusted_registrations"][period]
+      current_period[:adjusted_registrations] += facility_data[:adjusted_registrations][period]
```

Comment:
The totals are already available from the data_for_facility, so I ended up not adding them to this PR at all. The plan has changed a bit since this PR was started. It was originally about adding those totals, which I had wrongly thought needed to be aggregated by size. Now this PR is more about removing unnecessary data fetching. The previous plan was to show stats for all sizes, but now we only want stats for the currently displayed sizes. So I've removed the extra data fetching. Also, I obviously rewrote a bunch of specs. This is not ideal, but this project is a bit in the air right now.